api/src/main/java/jakarta/data/metamodel/FilterableAttribute.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel;
+
+import jakarta.data.Criteria;
+
+import java.util.Collection;
+
+/**
+ * Represents an entity attribute in the {@link StaticMetamodel} that supports filtering conditions.
+ * This interface provides methods for defining common query conditions, allowing the attribute to be used
+ * in filtering expressions such as `_Book.title.like("Jakarta Data%")`.
+ *
+ * <p>Example usage:</p>
+ * <pre>{@code
+ * // Define criteria for filtering
+ * Criteria titleCriteria = _Book.title.like("Jakarta Data%");
+ * Criteria dateCriteria = _Book.publicationDate.between(pastDate, LocalDate.now());
+ * Criteria authorCriteria = _Book.author.equal("John Doe");
+ *
+ * // Apply criteria in a repository query
+ * repo.books(List.of(titleCriteria, dateCriteria, authorCriteria));
+ * }</pre>
+ *
+ * @param <T> the type of the entity class in the static metamodel.
+ * @param <V> the type of the attribute's value.
+ */
+public interface FilterableAttribute<T, V> extends Attribute<T> {
```

Comment:
I would love to have more specializations, such as Boolean, with `isTrue` and `isFalse`methods on `String` `contains` and `like` and so on.

Reply:
JPQL doesn't currently have `is true` and `is false` operators. I think you're supposed to write `= true` and `= false`. So here you would just write `_Entity.booleanField.equals(true)`, for example.

JPQL *does* have `is null`, which looks like it's missing.

Reply:
I mean, by abstractions.

```java
_Car.available.isTrue();
```

The metadata will do it properly as you said.